implementations/rust/ockam/ockam_api/src/cli_state/storage/vaults_repository_sql.rs

```diff
@@ -27,43 +28,57 @@ impl VaultsSqlxDatabase {
 
 #[async_trait]
 impl VaultsRepository for VaultsSqlxDatabase {
-    async fn store_vault(&self, name: &str, path: &Path, is_kms: bool) -> Result<NamedVault> {
-        let query = query("INSERT INTO vault VALUES (?1, ?2, ?3, ?4)")
-            .bind(name.to_sql())
-            .bind(path.to_sql())
-            .bind(true.to_sql())
-            .bind(is_kms.to_sql());
-        query.execute(&*self.database.pool).await.void()?;
-
-        Ok(NamedVault::new(name, path.into(), is_kms))
+    async fn store_vault(&self, name: &str, vault_type: VaultType) -> Result<NamedVault> {
+        let mut transaction = self.database.begin().await.into_core()?;
+
+        let query1 =
+            query_scalar("SELECT EXISTS(SELECT 1 FROM vault WHERE is_default = $1)").bind(true);
+        let default_exists: Boolean = query1.fetch_one(&mut *transaction).await.into_core()?;
+        let default_exists = default_exists.to_bool();
```

Comment:
I need to check it, but now the "is_default" check is done here directly instead of in a higher layer?

Reply:
That was done incorrectly and, as a result, all the vaults were marked as default.